crates/core/src/kernel/transaction/conflict_checker.rs

```diff
@@ -203,21 +198,22 @@ impl<'a> TransactionInfo<'a> {
 
         if let Some(predicate) = &self.read_predicates {
             Ok(Either::Left(
-                files_matching_predicate(self.read_snapshot, &[predicate.clone()]).map_err(
-                    |err| CommitConflictError::Predicate {
+                files_matching_predicate(self.read_snapshot.clone(), &[predicate.clone()])
+                    .map_err(|err| CommitConflictError::Predicate {
                         source: Box::new(err),
-                    },
-                )?,
+                    })?,
             ))
         } else {
-            Ok(Either::Right(std::iter::empty()))
+            Ok(Either::Right(
+                self.read_snapshot.iter().map(|f| f.add_action()),
+            ))
```

Comment:
This may have been a bug in our conflict resolution. in the non-datafusion case, we always return all files to have been read from the table.

The previous logic would make sense if we guarantee that an appropriate predicate is always set when we read from the table (i.e. const eval to `true`), however I could find no place where we would be able to do so. As such it is at least more conservative to always return all files in the df case when there is no read predicate.

As we adopt more kernel tools also for conflict resolution, I plan to revisit all our logic, since we can do file skipping also in the non-df case.